File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/events/TaskListEvent.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esofthead.mycollab.module.project.events;

import com.esofthead.mycollab.vaadin.events.ApplicationEvent;

/**
 * 
 * @author haiphucnguyen
 */
public class TaskListEvent {

	public static class GotoRead extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoRead(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoEdit extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoEdit(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoAdd extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoAdd(Object source, Object data) {
			super(source, data);
		}
	}

	public static class ReoderTaskList extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public ReoderTaskList(Object source, Object data) {
			super(source, data);
		}
	}

	public static class SaveReoderTaskList extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public SaveReoderTaskList(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoGanttChartView extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoGanttChartView(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoTaskListScreen extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoTaskListScreen(Object source, Object data) {
			super(source, data);
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/project/events/TaskListEvent.scala
package com.esofthead.mycollab.module.project.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object TaskListEvent {

  class GotoGanttChartView(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class ReoderTaskList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class SaveReoderTaskList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoTaskListScreen(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
}
